client/src/views/MainView/MainView.ts

```diff
@@ -1,21 +1,203 @@
+    this.setCashbookList($cashbookList);
+  }
 
   template(): string {
-    return '';
+    return `
+      <div class="main-view">
+        <div class="input-bar"></div>
+        <div class="content">
+          <div class="header">
+            <label class="total-elements">전체 내역 ${this.totalElement}건</label>
+            <div class="selector">
+              <input id="checkbox-income" type="checkbox" hidden ${this.state.incomeChecked ? 'checked' : ''}>
+              <label for="checkbox-income" class="checkmark"></label>
+              <label class="income">수입 ${this.totalIncome.toLocaleString()}</label>
+
+              <input id="checkbox-expenditure" type="checkbox" hidden ${this.state.expenditureChecked ? 'checked' : ''}>
+              <label for="checkbox-expenditure" class="checkmark"></label>
+              <label class="expenditure">지출 ${this.totalExpenditure.toLocaleString()}</label>
+            </div>
+          </div>
+          <ul class="cashbook-list"></ul>
+        </div>
+      </div>
+    `;
+  }
+
+  setEvent() {
+    const $incomeCheckbox = this.$target.querySelector('#checkbox-income') as HTMLInputElement;
+    const $expenditureCheckbox = this.$target.querySelector('#checkbox-expenditure') as HTMLInputElement;
+
+    $incomeCheckbox?.addEventListener('change', () => {
+      this.setState({
+        incomeChecked: $incomeCheckbox.checked,
+      });
+    });
+    $expenditureCheckbox?.addEventListener('change', () => {
+      this.setState({
+        expenditureChecked: $expenditureCheckbox.checked,
+      });
+    });
   }
 
-  setEvent() {}
+  setCashbookData(year?: number, month?: number) {
+    // 서버 요청
+    const result = dummy;
+    this.totalElement = result.totalElements;
+    this.totalIncome = result.totalIncome;
+    this.totalExpenditure = result.totalExpenditure;
+    this.cashbooks = result.cashbooks;
+  }
+
+  getMonth() {
+    return this.state.month;
+  }
+
+  setCashbookList($cashbookList: HTMLUListElement) {
+    this.registerCashbookToList($cashbookList);
+  }
+
+  registerCashbookToList($list: HTMLUListElement) {
+    const year = this.state.year as number;
+    let month = String(this.state.month);
+    const { incomeChecked, expenditureChecked } = this.state;
+    if (Number(month) < 10) month = '0' + month;
+    for (let i = 31; i >= 1; i--) {
+      const cashbooks = this.cashbooks
+        .filter((cashbook) => cashbook.date === `${year}-${month}-${i}`)
+        .filter(
+          (cashbook) =>
+            (cashbook.categoryType === 'income' && incomeChecked) ||
+            (cashbook.categoryType === 'expenditure' && expenditureChecked)
+        );
+      if (cashbooks.length === 0) continue;
+      const dayIndex = new Date(year, Number(month) - 1, i).getDay();
+      const totalIncome: number = cashbooks
+        .filter((cashbook) => cashbook.categoryType === 'income')
+        .map((cashbook) => cashbook.price)
+        .reduce((a, b) => a + b, 0);
```

Comment:
되게 좋은 코드인 것 같습니다! 다만 배열 순환이 많다보니, 
reduce안에서 조건문 처리하는 방법을 이용해서 한번에 처리하는 방법도 괜찮을 것 같아서 적어놓습니다!
```suggestion
      const totalIncome: number = cashbooks
        .reduce((a, cashbook) => { 
                   if(cashbook.categoryType === 'income') 
                        return cashbook.price + a; 
                    return a;
        }, 0);
```